cookbook/lvgl.rst

```diff
@@ -0,0 +1,1655 @@
+                    id: lvgl_led
+                    color: 0xFF0000
+
+A few notable things in this example: usage of a base object ``obj`` as a parent for the label (in order to center the label in the middle of it and emphasize it with shadows independently of the label's dimensions); usage of ``align_to`` to align it to the led vertically; changing the background color of the buttons in ``pressed`` state; using the ``key_code`` configuration option to send a different character to ``key_collector`` instead of the displayed symbol.
+
+.. _lvgl-cook-idlescreen:
+
+Turn off screen when idle
+-------------------------
+
+LVGL has a notion of screen inactivity, i.e. how long did the user not interact with the screen. This can be used to dim the display backlight or turn it off after a moment of inactivity (like a screen saver). Touching the screen counts as an activity and resets the inactivity counter (it's important to use the ``on_release`` trigger). With a template number you can make the timeout settable by the users.
+
+.. code-block:: yaml
+
+    lvgl:
+      ...
+      on_idle:
+        timeout: !lambda "return (id(display_timeout).state * 1000);"
+        then:
+          - logger.log: "LVGL is idle"
+          - light.turn_off: display_backlight
+          - lvgl.pause:
+
+    touchscreen:
+      - platform: ...
+        on_release:
+          - if:
+              condition: lvgl.is_paused
+              then:
+                - logger.log: "LVGL resuming"
+                - lvgl.resume:
+                - lvgl.widget.redraw:
+                - light.turn_on: display_backlight
+
+    light:
+      - platform: ...
+        id: display_backlight
+
+    number:
+      - platform: template
+        name: LVGL Screen timeout
+        optimistic: true
+        id: display_timeout
+        unit_of_measurement: "s"
+        initial_value: 45
+        restore_value: true
+        min_value: 10
+        max_value: 180
+        step: 5
+        mode: box
+
+.. _lvgl-cook-antiburn:
+
+Prevent burn-in of LCD
+----------------------
+
+You can use this to protect and prolong the lifetime of the LCD screens, thus being more green and generating less hazardous waste.
+
+Wall mounted LCD screens' main problem is that they display the same picture 99.999% of the time. Even if somebody turns off backlight during the night or dark periods, the LCD screen keeps showing the same picture, seen by nobody. There are high chances that this will lead to screen picture burn-in after a few years of operation.
+
+One way to mitigate this is to *train* the pixels periodically with completely different other content. ``show_snow`` option during LVGL paused state was developed in this scope, to  display random colored pixels across the entire screen in order to minimize screen burn-in, to relief the tension put on each individual pixel.
+
+In the example below pixel training is done four times for a half an hour every night, can also be stopped by touching the screen.
```

Comment:
```suggestion
In the example below, pixel exercising is done four times for a half an hour every night; it can be stopped by touching the screen.
```

Reply:
The technical term in the display industry is "pixel training", not exercising.